src/urlschemehandler.cpp

```diff
@@ -216,6 +218,33 @@ UrlSchemeHandler::replyZimNotFoundPage(QWebEngineUrlRequestJob *request,
     request->reply("text/html", buffer);
 }
 
+void
+UrlSchemeHandler::replyBadZimFilePage(QWebEngineUrlRequestJob *request,
+                                  const QString &zimId)
+{
+    QBuffer *buffer = new QBuffer;
+    const auto& book = KiwixApp::instance()->getLibrary()->getBookById(zimId);
+    const QString path = QString::fromStdString(book.getPath());
+    const QString name = QString::fromStdString(book.getName());
+    const QString zimEntryPath = request->requestUrl().path();
+
+    QString contentHtml = "<section><div>"
+                          "<h1>" + gt("bad-zim-file-error-page-title") + "</h1>"
+                          "<p>"  + gt("bad-zim-file-error-page-text") + "</p>"
+                          "<p>"  + gt("zim-id")   + ": <b>" + zimId + "</b></p>"
+                          "<p>"  + gt("zim-name") + ": <b>" + name  + "</b></p>"
+                          "<p>"  + gt("zim-path") + ": <b>" + path  + "</b></p>"
+                          "<p>" +  gt("zim-entry-path") + ": <b>" + zimEntryPath + "</b></p>"
+                          "</div></section>";
```

Comment:
Shouldn't we put all this text as a translatable template ?
RTL language may need to be formatted differently.

Reply:
This problem was introduced in #1132, I only duplicated it. Maybe we can address it in a separate PR.

Reply:
I don't know about RTL, but the current way of doing is the one recommended by TW.

Reply:
The concern is about "*key*: *value*" lines. How will they be displayed if *key* is in an RTL language/script and *value* is in a LTR script?

Reply:
Honestly pretty sure it will be all-right. Merging and we will fix if not, but this is anyway a general problem then.